src/xds.rs

```diff
@@ -54,6 +55,22 @@ pub enum Error {
     OnDemandSend(),
 }
 
+/// DeleteReason enumerates the possible reasons that ztunnel is removing a workload from proxy state
+pub enum DeleteReason {
+    XdsRemove,
+    /// XdsPreClean means we're inserting a workload which involves deleting the old version of the workload to ensure a clean insert
+    XdsPreClean,
+}
```

Comment:
About this, i would suggest not making it complex

Reply:
@howardjohn

Reply:
Yeah I mostly agree this seems a bit excessive just for debug logs

Reply:
Are we OK with every update logging about the update and the delete so long as it's at debug?

Reply:
Yes, sounds good. Workload is a fast-changing resource